### ISSUE TITLE
cc1plus: some warnings being treated as errors

### ISSUE THREAD

**Issue by jiapei100** [OpenImageIO/oiio#2141]
```
....../oiio/src/libOpenImageIO/imagebufalgo_opencv.cpp:286:14: error: variable ‘ok’ set but not used [-Werror=unused-but-set-variable]
         bool ok = true;
              ^~
cc1plus: some warnings being treated as errors


....../oiio/src/include/OpenImageIO/imagebufalgo_util.h:213:42: error: lvalue required as left operand of assignment
         ret = func<float> (R, __VA_ARGS__); break;                      \
                                          ^
....../oiio/src/libOpenImageIO/imagebufalgo_opencv.cpp:287:9: note: in expansion of macro ‘OIIO_DISPATCH_TYPES’
         OIIO_DISPATCH_TYPES(true, "from_OpenCV R/B swap", RBswap, dstformat, dst,

cc1plus: some warnings being treated as errors
```

Failed to build **oiio** with current **OpenCV4** ???

Looks like it has **SOMETHING** to do with the **MACRO** in file **imagebufalgo_util.h** .

```
// Macro to call a type-specialzed version func<type>(R,...)
#define OIIO_DISPATCH_TYPES(ret,name,func,type,R,...)                   \
    switch (type.basetype) {                                            \
    case TypeDesc::FLOAT :                                              \
        ret = func<float> (R, __VA_ARGS__); break;                      \
    case TypeDesc::UINT8 :                                              \
        ret = func<unsigned char> (R, __VA_ARGS__); break;              \
    case TypeDesc::HALF  :                                              \
        ret = func<half> (R, __VA_ARGS__); break;                       \
    case TypeDesc::UINT16:                                              \
        ret = func<unsigned short> (R, __VA_ARGS__); break;             \
    case TypeDesc::INT8  :                                              \
        ret = func<char> (R, __VA_ARGS__); break;                       \
    case TypeDesc::INT16 :                                              \
        ret = func<short> (R, __VA_ARGS__); break;                      \
    case TypeDesc::UINT  :                                              \
        ret = func<unsigned int> (R, __VA_ARGS__); break;               \
    case TypeDesc::INT   :                                              \
        ret = func<int> (R, __VA_ARGS__); break;                        \
    case TypeDesc::DOUBLE:                                              \
        ret = func<double> (R, __VA_ARGS__); break;                     \
    default:                                                            \
        (R).error ("%s: Unsupported pixel data format '%s'", name, type); \
        ret = false;                                                    \
    }
```

**Comment by lgritz** [comment]
Platform? Compiler?

**Comment by jiapei100** [comment]
@lgritz 

```
➜  ~ uname -r
4.19.13-041913-generic
➜  ~ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.1 LTS
Release:        18.04
Codename:       bionic
➜  ~ gcc --version
gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by lgritz** [comment]
Proposed fix in #2143. Can you take a look at that and give it a try on your end to confirm that it fixes for you?

**Comment by lgritz** [comment]
By the way, on any build, you can set the build flag `STOP_ON_WARNING=0` to let it no longer consider warnings to be errors. That would also have gotten you unstuck (though I'm glad you pointed out the problem so we can fix it more permanently).
